src/main/java/org/jenkinsci/plugins/registry/notification/events/EventType.java

```diff
@@ -28,7 +29,10 @@ public void buildEnvironment(@Nonnull EnvVars envs, @Nonnull PushNotification pu
                 JSONObject contents = data.getJSONObject("contents");
                 if (contents != null) {
                     DateTimeFormatter parser = ISODateTimeFormat.dateTimeParser();
-                    push.setPushedAt(parser.parseDateTime(getTimeStamp(contents)).toDate());
+                    String timeStamp = getTimeStamp(contents);
+                    if( timeStamp != null ) {
+                        push.setPushedAt(parser.parseDateTime(timeStamp).toDate());
```

Comment:
`getTimestamp` could probably do the parsing to a date as well and return `null` if the string couldn't be parsed.
http://www.joda.org/joda-time/apidocs/org/joda/time/format/DateTimeFormatter.html#parseDateTime-java.lang.String-

Reply:
though it doesn't explicitly say so, it seems like `org.joda.time.format.DateTimeFormatter` is thread safe, so you could have it as a shared constant if you feel creating a new one each time is bad.

Reply:
Hey @rsandell - finally back to looking at your comments here.  Are you thinking that pushing the Parser down to the child classes is better than having it in the parent?  That's either going to duplicate that code in each subclass or end up just pushing 2 or 3 lines in this method to a protected method elsewhere in the EventType.java class.

Reply:
It was such a long time ago that I can't remember :) Will need to rereview :)